walk/git.go

```diff
@@ -38,7 +38,9 @@ func (g *GitReader) Read(ctx context.Context, files []*File) (n int, err error)
 		r, w := io.Pipe()
 
 		// create a command which will execute from the specified sub path within root
-		cmd := exec.Command("git", "ls-files")
+		// we pass `-c` to pick up any files in the git index (this is the default ls-files behaviour)
+		// we pass `-o` to pick up untracked files
+		cmd := exec.Command("git", "ls-files", "-co")
```

Comment:
nit: IMO, using the long form of these options (`--cached` and `--ignored`) would be slightly more readable and reduce the need for the documentation explaining the options.